### ISSUE TITLE
Accuracy is NaN if dataset is small

### ISSUE THREAD

**Issue by StpMax** [mindsdb/lightwood#876]
If training dataset is small, then result predictor accuracy is NaN

## How can we replicate it?
Make connection to test postgres database with `home_rentals` dataset, train two predictors:
```
create predictor t1_rp from pgpg (select * from test_data.home_rentals limit 1000) predict rental_price;
create predictor t2_rp from pgpg (select * from test_data.home_rentals limit 10) predict rental_price;
```
`r2_score` for `t1_rp` is 0.999
`r2_score` for `t2_rp` is NaN

**Comment by paxcema** [comment]
Underlying cause: `sklearn.metrics.r2_score` receives a single sample, and it's undefined for this edge case. We can probably tune the splitter so that at least two data points make their way into the validation dataset.
